mslib/msui/mpl_pathinteractor.py

```diff
@@ -472,15 +428,508 @@ def draw_callback(self, event):
             # (see infos on http://www.scipy.org/Cookbook/Matplotlib/Animations).
             # self.canvas.blit(self.ax.bbox)
 
+    def set_vertices_visible(self, showverts=True):
+        """Set the visibility of path vertices (the line plot).
+        """
+        self.showverts = showverts
+        self.line.set_visible(self.showverts)
+        for t in self.wp_labels:
+            t.set_visible(showverts and self.label_waypoints)
+        if not self.showverts:
+            self._ind = None
+        self.canvas.draw()
+
+    def set_patch_visible(self, showpatch=True):
+        """Set the visibility of path patch (the area).
+        """
+        self.pathpatch.set_visible(showpatch)
+        self.canvas.draw()
+
+    def set_labels_visible(self, visible=True):
+        """Set the visibility of the waypoint labels.
+        """
+        self.label_waypoints = visible
+        for t in self.wp_labels:
+            t.set_visible(self.showverts and self.label_waypoints)
+        self.canvas.draw()
+
+    def redraw_path(self, vertices=None, waypoints_model_data=[]):
```

Comment:
empty list is  passing mutable default arguments to functions
https://towardsdatascience.com/18-common-python-anti-patterns-i-wish-i-had-known-before-44d983805f0f